server/queue/queue-api/src/test/java/org/apache/james/queue/api/ManageableMailQueueContract.java

```diff
@@ -457,24 +454,25 @@ default void removeShouldNotFailWhenBrowsingIterating() throws Exception {
 
     @Test
     default void browseShouldNotFailWhenConcurrentRemoveWhenIterating() throws Exception {
-        enQueue(defaultMail()
-            .name("name1")
-            .build());
-        enQueue(defaultMail()
-            .name("name2")
-            .build());
-        enQueue(defaultMail()
-            .name("name3")
-            .build());
+        // We use a large number of emails so that the probability of the remove being propagated
+        // through the queue is high enough that the test is not flaky.
+        IntStream.range(0, 100).forEach(
+                Throwing.intConsumer(i ->
+                        enQueue(defaultMail()
+                                .name("name" + i)
+                                .build())
+                ).sneakyThrow()
+        );
 
         ManageableMailQueue.MailQueueIterator items = getManageableMailQueue().browse();
-        items.next();
+        items.next(); // we consume 1 here
 
-        getManageableMailQueue().remove(ManageableMailQueue.Type.Name, "name2");
+        getManageableMailQueue().remove(ManageableMailQueue.Type.Name, "name99");
 
-        awaitRemove();
 
-        assertThatCode(() ->  Iterators.consumingIterator(items)).doesNotThrowAnyException();
+        assertThatCode(
+                () -> assertThat(consumeIterator(items)).isEqualTo(98) // and 998 here so 999 in total
+        ).doesNotThrowAnyException();
```

Comment:
interestingly, this test fails except for the pulsar implementation. 
Most implementations will return the removed item if the browse has been started before the item is removed. The pulsar implementation is able to filter it out even from an operating iterator. 

I am considering lowering the constraint here with this code
```
        assertThatCode(
                // the remove may or may not be applied to an already started iterator
                // as long as it doesn't make the iterator crash
                () -> assertThat(consumeIterator(items)).isGreaterThanOrEqualTo(98).isLessThan(100)
        ).doesNotThrowAnyException();
```

Reply:
So the over implementation "browse" reads from a snapshot while here deletion are dynamically applied onto the "browsing side. Ok with me.

I am ok with the isolation level of the browse to be implementation specific.

We could likely benefit from a little write up regarding this? Interesting details could be documented...

Reply:
> We could likely benefit from a little write up regarding this? Interesting details could be documented...

I'm ok to give it a try, where would you expect that : in the API doc of the ManagedMailqueue, in a page of the website (new/old) or on the ML ?